Compatibility with boon table plugin

knoxfighter opened the pull request:
- added calls to options_windows()
- added e8 export
- added Log window, to show what is printed with e3/e8
- added e5 export for default colors

knoxfighter committed:
fixed some errors occuring since my setup is more strict

knoxfighter committed:
added calls to options_windows

knoxfighter committed:
added working e3/e8 logging with Logs window

knoxfighter committed:
use list instead of deque

knoxfighter committed:
added e5() export for colors (colors are the default colors of arcdps)

knoxfighter commented on arcdps_mock/main.cpp:
```diff
@@ -43,9 +47,111 @@ struct ArcModifiers
+	ImVec4(1.000000f,0.800000f,0.000000f,0.210000f),
+	ImVec4(0.000000f,0.800000f,1.000000f,0.210000f),
+	ImVec4(1.000000f,0.470000f,0.800000f,0.210000f)
+};
```
If you want this definitions in a different place, just say so (it not yet had a file just for global objects like this=

Krappa322 replied:
nah this is fine. The arrays can all be marked const and static though?

knoxfighter replied:
sure

Krappa322 commented on arcdps_mock/CombatMock.h:
```diff
@@ -68,8 +76,9 @@ class CombatMock
 	void DisplayActions();
+	void DisplayLog();
 
-	const arcdps_exports* const myCallbacks;
```
Why remove these const and the const in the constructor? I can't see where you are modifying the struct (or why you would want to do so)

knoxfighter replied:
i had to, else i was unable to compile it, i can try to add it back in again

Krappa322 replied:
I think the reason you couldn't compile was because your new constructor is leaving myCallbacks with an undefined value

Krappa322 commented on arcdps_mock/CombatMock.h:
```diff
@@ -45,7 +47,8 @@ class CombatMock
 
 public:
-	CombatMock(const arcdps_exports* pCallbacks);
+	CombatMock(){}
```
What is the meaning if initializing it like this? The objects initialized like this are just unusable?

I think the object in main.cpp should just be a pointer instead

knoxfighter replied:
ok, a pointer it is, i would guessa unique_ptr?

Krappa322 replied:
yeah unique_ptr seems good

knoxfighter replied:
I created the new constructor after i had problems to compile, i will revert that and see if it still not works

Krappa322 commented on arcdps_mock/main.cpp:
```diff
@@ -43,9 +47,111 @@ struct ArcModifiers
+	ImVec4(0.250000f,0.220000f,0.230000f,1.000000f),
+	ImVec4(0.000000f,0.000000f,5.688562f,-0.000000f)
+};
+ImVec4 profColsBase[9] { // size of enum `Prof`
```
Just a thought:

I usually add a dummy value "Max" at the end of all enums. So here you could define profColsBase[PROF_MAX] 

This hit me with the stuff in arcdps-extension as well since you're passing an initializer list of all values there. If you had _MAX for your enums then you could just send in the enum max value instead of having to specify every value in the enum (and the function iterates between 0 and _MAX).

Krappa322 replied:
>the stuff in arcdps-extension

Referring to https://github.com/knoxfighter/arcdps-extension/commit/e647aaad52978420d33a090b32b9c476865c9916

knoxfighter replied:
I could do that, but i sometimes only want to have a few of the available enum values selectable :)
So an overload is the way to go ^^

knoxfighter replied:
I will do that tomorrow :)

Krappa322 commented on arcdps_mock/main.cpp:
```diff
@@ -161,10 +272,21 @@ int Run(const char* pModulePath, const char* pMockFilePath)
+				TEST_MODULE_EXPORTS.options_windows("log");
+				TEST_MODULE_EXPORTS.options_windows("bufftable");
+				TEST_MODULE_EXPORTS.options_windows("Errors");
+				TEST_MODULE_EXPORTS.options_windows(0);
```
nullptr? :)

knoxfighter replied:
yes, arcdps is calling it with a nullptr at the bottom, so we can have all the checkboxes on the top :)

Krappa322 replied:
I just meant to use nullptr instead of 0 haha

knoxfighter committed:
made color arrays const static

knoxfighter committed:
reverted the change i made to make it compilable, they are not needed anymore

knoxfighter committed:
Merge remote-tracking branch 'krappa322/master' into develop

knoxfighter committed:
use FINAL_ENTRY of arcdps enums

knoxfighter committed:
create directory 'addons/arcdps' it is needed for plugins to dsave settings

knoxfighter committed:
Merge remote-tracking branch 'krappa322/master' into develop

knoxfighter commented:
I have changed what you wanted and also added the creation of `addons/arcdps` folders, so my plugin can save it's settings :)
If you have anything else say so, otherwise it is ready to get merged.
